ktor-io/jvm/src/io/ktor/utils/io/jvm/javaio/Blocking.kt

```diff
@@ -246,6 +244,21 @@ private abstract class BlockingAdapter(val parent: Job? = null) {
     @OptIn(InternalCoroutinesApi::class)
     private fun parkingLoop(thread: Thread) {
         if (state.value !== thread) return
+        if (!isParkingAllowed()) {
+            ADAPTER_LOGGER.warn(
+                """
+                Blocking network thread detected. 
+                Please make sure you're using blocking IO primitives like InputStream and OutputStream only only in 
+                the context of Dispatchers.IO:
+                ```
+                withContext(Dispatchers.IO) {
+                    myInputStream.read()
+                }
+                ```
+                It can possible lead to a performance decline or even a deadlock.
```

Comment:
This line is confusing because it is placed after the solution to the problem (feels like the solution can lead to a decline or a deadlock). It's better to structure it this way: problem description -> possible consequences -> the solution